src/event-stream/event-server.ts

```diff
@@ -692,18 +696,35 @@ function createMessageProcessorQueue(): EventMessageHandler {
   // Create a promise queue so that only one message is handled at a time.
   const processorQueue = new PQueue({ concurrency: 1 });
 
+  const eventTimer = new prom.Histogram({
+    name: 'event_ingestion_timers',
+    help: 'Event ingestion timers',
+    labelNames: ['event'],
+    buckets: prom.exponentialBuckets(1, 2, 22), // 22 buckets, from 1ms to 35minutes
+  });
```

Comment:
@CharlieC3 how does this prometheus metric look? Here's some example output:
```
# HELP event_ingestion_timers Event ingestion timers
# TYPE event_ingestion_timers histogram
event_ingestion_timers_bucket{le="1",event="block"} 0
event_ingestion_timers_bucket{le="2",event="block"} 0
event_ingestion_timers_bucket{le="4",event="block"} 0
event_ingestion_timers_bucket{le="8",event="block"} 0
event_ingestion_timers_bucket{le="16",event="block"} 0
event_ingestion_timers_bucket{le="32",event="block"} 29
event_ingestion_timers_bucket{le="64",event="block"} 33
event_ingestion_timers_bucket{le="128",event="block"} 34
event_ingestion_timers_bucket{le="256",event="block"} 35
event_ingestion_timers_bucket{le="512",event="block"} 35
event_ingestion_timers_bucket{le="1024",event="block"} 35
event_ingestion_timers_bucket{le="2048",event="block"} 35
event_ingestion_timers_bucket{le="4096",event="block"} 35
event_ingestion_timers_bucket{le="8192",event="block"} 35
event_ingestion_timers_bucket{le="16384",event="block"} 35
event_ingestion_timers_bucket{le="32768",event="block"} 35
event_ingestion_timers_bucket{le="65536",event="block"} 35
event_ingestion_timers_bucket{le="131072",event="block"} 35
event_ingestion_timers_bucket{le="262144",event="block"} 35
event_ingestion_timers_bucket{le="524288",event="block"} 35
event_ingestion_timers_bucket{le="1048576",event="block"} 35
event_ingestion_timers_bucket{le="2097152",event="block"} 35
event_ingestion_timers_bucket{le="+Inf",event="block"} 35
event_ingestion_timers_sum{event="block"} 1119
event_ingestion_timers_count{event="block"} 35
event_ingestion_timers_bucket{le="1",event="raw_event"} 145
event_ingestion_timers_bucket{le="2",event="raw_event"} 153
event_ingestion_timers_bucket{le="4",event="raw_event"} 162
event_ingestion_timers_bucket{le="8",event="raw_event"} 167
event_ingestion_timers_bucket{le="16",event="raw_event"} 170
event_ingestion_timers_bucket{le="32",event="raw_event"} 171
event_ingestion_timers_bucket{le="64",event="raw_event"} 172
event_ingestion_timers_bucket{le="128",event="raw_event"} 172
event_ingestion_timers_bucket{le="256",event="raw_event"} 172
event_ingestion_timers_bucket{le="512",event="raw_event"} 172
event_ingestion_timers_bucket{le="1024",event="raw_event"} 172
event_ingestion_timers_bucket{le="2048",event="raw_event"} 172
event_ingestion_timers_bucket{le="4096",event="raw_event"} 172
event_ingestion_timers_bucket{le="8192",event="raw_event"} 172
event_ingestion_timers_bucket{le="16384",event="raw_event"} 172
event_ingestion_timers_bucket{le="32768",event="raw_event"} 172
event_ingestion_timers_bucket{le="65536",event="raw_event"} 172
event_ingestion_timers_bucket{le="131072",event="raw_event"} 172
event_ingestion_timers_bucket{le="262144",event="raw_event"} 172
event_ingestion_timers_bucket{le="524288",event="raw_event"} 172
event_ingestion_timers_bucket{le="1048576",event="raw_event"} 172
event_ingestion_timers_bucket{le="2097152",event="raw_event"} 172
event_ingestion_timers_bucket{le="+Inf",event="raw_event"} 172
event_ingestion_timers_sum{event="raw_event"} 313
event_ingestion_timers_count{event="raw_event"} 172
event_ingestion_timers_bucket{le="1",event="burn_block"} 0
event_ingestion_timers_bucket{le="2",event="burn_block"} 1
event_ingestion_timers_bucket{le="4",event="burn_block"} 115
event_ingestion_timers_bucket{le="8",event="burn_block"} 131
event_ingestion_timers_bucket{le="16",event="burn_block"} 135
event_ingestion_timers_bucket{le="32",event="burn_block"} 137
event_ingestion_timers_bucket{le="64",event="burn_block"} 137
event_ingestion_timers_bucket{le="128",event="burn_block"} 137
event_ingestion_timers_bucket{le="256",event="burn_block"} 137
event_ingestion_timers_bucket{le="512",event="burn_block"} 137
event_ingestion_timers_bucket{le="1024",event="burn_block"} 137
event_ingestion_timers_bucket{le="2048",event="burn_block"} 137
event_ingestion_timers_bucket{le="4096",event="burn_block"} 137
event_ingestion_timers_bucket{le="8192",event="burn_block"} 137
event_ingestion_timers_bucket{le="16384",event="burn_block"} 137
event_ingestion_timers_bucket{le="32768",event="burn_block"} 137
event_ingestion_timers_bucket{le="65536",event="burn_block"} 137
event_ingestion_timers_bucket{le="131072",event="burn_block"} 137
event_ingestion_timers_bucket{le="262144",event="burn_block"} 137
event_ingestion_timers_bucket{le="524288",event="burn_block"} 137
event_ingestion_timers_bucket{le="1048576",event="burn_block"} 137
event_ingestion_timers_bucket{le="2097152",event="burn_block"} 137
event_ingestion_timers_bucket{le="+Inf",event="burn_block"} 137
event_ingestion_timers_sum{event="burn_block"} 565
event_ingestion_timers_count{event="burn_block"} 137
```

Reply:
Excited to see we're getting these metrics soon! A couple comments:
* Mind if we rename it to add a prefix, like `stacks_event_ingestion_timers_...`? Denoting the metric's relation to stacks helps when searching for stacks-related metrics amongst the tens of thousands of other metrics we have.
* Can we reduce the cardinality of the bucket metrics? I don't think we need so much accuracy between buckets, because the downside for this is pretty significant on performance in prometheus and mimir.
    * For example, le=1 and le=2 are very close to each other and denoting the difference between them doesn't yield much value to anyone observing it.
    * If we can reduce the cardinality to something like 10 buckets per bucket metric, that would help a ton at keeping our systems (or whosever runs this) performant.

Reply:
Done, for both points!

Reply:
Perfect, thank you!